### DISCUSSION TITLE
When using the custom page `StarlightPage`, how to manage two `<html>` labels?

### DISCUSSION THREAD

**Discussion by sgalcheung** [withastro/starlight#2585] [Q&A]
> links [Disscussion #2389]( #https://github.com/withastro/starlight/discussions/2389)

When I used a custom page to make the starlight layout as a part of the website, here are two `<html>` pages on my website.

The root website at https://starlight-squidex.netlify.app/ has a `<html>` page.
![image](https://github.com/user-attachments/assets/9a07b6e9-543b-490f-bb7d-8dac30cb5168)

And the Starlight page has another `<html>` page. you can visit it at https://starlight-squidex.netlify.app/column/article/c47961ee-ff4c-4fbb-b335-017487e75377
![image](https://github.com/user-attachments/assets/35b1c64e-d80a-4e38-bd6b-268c3d4de708)

**Comment by HiDeoo** [comment]
If you're creating a custom page, e.g. `src/pages/custom.astro`, you only have to wrap your content with the `<StarlightPage />` component.

```astro
---
// src/pages/custom.astro
import StarlightPage from '@astrojs/starlight/components/StarlightPage.astro';
---

<StarlightPage frontmatter={{ title: 'My custom page' }}>
  <p>This is the custom page content.</p>
</StarlightPage>
```

There is no need to also include your own `<html>` element or layout in the custom page. The `<StarlightPage />` component will take care of that for you.

**Comment by sgalcheung** [comment]
In my case(as above), the docs(starlight website) is a part of the website, my website has an `html` element, and I only want the `<StarlightPage>`  as a component, but it also has an `html` element.

I read the source code of Starlight, the `<StarlightPage>` base on `<Page>`, and `<Page>` is a component implemented by Page.astro, and it has an `html` element, so I just want it maybe have a config to set another label or disable it.

**Comment by sgalcheung** [comment]
Since Starlight released version [0.32](https://astro.build/blog/starlight-032/), it supports route middleware,  we can better handle this, there is an example of this to override the Starlight page, [Custom page component](https://starlight-examples.netlify.app/examples/page-override/).
